src/test/java/com/appdirect/sdk/credentials/SourceBackedCredentialsTest.java

```diff
@@ -0,0 +1,50 @@
+package com.appdirect.sdk.credentials;
+
+import com.appdirect.sdk.appmarket.Credentials;
+
+import java.util.function.Function;
+
+import org.assertj.core.api.Condition;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class SourceBackedCredentialsTest {
+
+
+	private Function<String, Credentials> functionTest = (String k) ->  {
+		if (k.equalsIgnoreCase("bad-key")) {
+			return Credentials.invalidCredentials();
+		}
+		return new Credentials(k,"s1");
+	};
+
+	@Test
+	public void returnsGoodCredentials_whenKeyIsFound() throws Exception {
+		//Given
+		SourceBackedCredentials findCredentials = new SourceBackedCredentials(functionTest);
+
+		//When
+		Credentials credentials = findCredentials.getConsumerCredentials("key1");
+
+		//Then
+		assertThat(credentials).is(credentialsOf("key1", "s1"));
```

Comment:
I must admit that it looks beautiful, however I'd prefer to assert with equality:
```suggestion
		assertThat(credentials).isEqualTo(expected)
```

Define also a Credentials object in the `Given` section with the key and value to be used in the function